types/src/checksummed_hex.rs

```diff
@@ -140,6 +139,18 @@ pub fn decode(input: &(impl AsRef<[u8]> + ?Sized)) -> Result<Vec<u8>, base16::De
     Ok(bytes)
 }
 
+/// Stuff
+pub trait ChecksummedHexEncode<const N: usize>: AsRef<[u8]> + AsMut<[u8]> {
+    /// More stuff
+    fn checksummed_hex_encode(&self) -> String {
+        encode_iter(self.as_ref()).collect()
+    }
+}
+
+impl ChecksummedHexEncode<8> for [u8; 8] {}
+impl ChecksummedHexEncode<16> for [u8; 16] {}
+impl ChecksummedHexEncode<32> for [u8; 32] {}
+impl ChecksummedHexEncode<64> for [u8; 64] {}
```

Comment:
Something like this should work

```suggestion
impl<N> ChecksummedHexEncode<N> for [u8; N] {}
```

That way it only monomorphizes over the array lengths we actually use.